openspec/changes/archive/2026-08-11-issue-496-async-http-refactor/design.md

```
@@ -0,0 +1,195 @@
+    #[error("failed to parse version: {0}")]
+    ParseError(String),
+}
+```
+
+**Alternatives considered**: Using `anyhow` for all errors — rejected; the success criteria requires categorizing errors (timeout vs. connection vs. status). `thiserror` gives structured variants for QA and logging.
+
+**Rationale**: Aligns with `SkillInstallError` in `install.rs` which already uses `thiserror` with `Network` variants. Structured errors make the acceptance criteria verifiable.
+
+## Data Flow
+
+```
```

markdownlint-cli2 (0.23.2)

[warning] Fenced code blocks should have a language specified

(MD040, fenced-code-language)